pkg/serving/v1/client.go

```diff
@@ -378,6 +378,13 @@ func getBaseRevision(cl KnServingClient, service *servingv1.Service) (*servingv1
 
 // Delete a revision by name
 func (cl *knServingClient) DeleteRevision(name string, timeout time.Duration) error {
+	revision, err := cl.client.Revisions(cl.namespace).Get(name, v1.GetOptions{})
+	if err != nil {
+		return clienterrors.GetError(err)
+	}
+	if revision.GetDeletionTimestamp() != nil {
+		return fmt.Errorf("can't delete revision %s because it has been marked for deletion", name)
```

Comment:
```suggestion
		fmt.Printf("\nWARNING: can't delete revision %s because it is already marked for deletion.\n\n", name)
		return nil
```

Reply:
Well, looking at https://github.com/knative/client/blob/c6537a8e842782225dfd3ec0a5281262aec50587/pkg/kn/commands/revision/delete.go#L55-L60

We're not exiting with non-zero exit code at all, so returning with `fmt.Errorf` should be okay as well.

Reply:
@MIBc : You can discard my comments above and follow suggestion given by @rhuss.